### PR TITLE
Feature/respawnable spawners

### PR DESCRIPTION
**Pull request by MichaelHillcox**
Adds the ability to configure a feature I'm calling `respawnable spawners`

When abled, at a specific interval, the game will watch for all broken spawners and respawn them after a configured interval. This will by default return the original spawners, but can be configured to use a list of entities to spawn from. All invalid entities are removed at load time, if there is no valid entities left, we default back to the original spawner. 

Let me know if FTB Lib can replace any of the systems I hacked into the bottom :D

### CODE REVIEW

File: common/src/main/java/dev/ftb/packcompanion/features/ServerFeature.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package dev.ftb.packcompanion.features;
```
Comment (desht) [inline common/src/main/java/dev/ftb/packcompanion/features/ServerFeature.java:1]: Feels like this class could be a subclass of CommonFeature to reduce duplication, but not a big deal

File: common/src/main/java/dev/ftb/packcompanion/features/buffs/MobEntityBuff.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+public class MobEntityBuff extends ServerFeature {
+    @Override
+    public void initialize() {
+        // On entity load
+        EntityEvent.ADD.register((entity, world) -> {
+            if (entity instanceof Enemy && entity instanceof Mob mob) {
+                AttributeInstance attribute = mob.getAttribute(Attributes.MAX_HEALTH);
+                if (attribute != null) {
+                    attribute.setBaseValue(attribute.getBaseValue() * PCServerConfig.MODIFY_MOB_BASE_HEALTH.get());
+                    ((Mob) entity).setHealth((float) attribute.getBaseValue());
```
Comment (desht) [inline common/src/main/java/dev/ftb/packcompanion/features/buffs/MobEntityBuff.java]: Nitpicking but `mob.setHealth(...)` would work here

Comment (MichaelHillcox) [reply common/src/main/java/dev/ftb/packcompanion/features/buffs/MobEntityBuff.java]: Oh, yeah, that's true... Whoops

File: common/src/main/java/dev/ftb/packcompanion/features/spawners/SpawnerManager.java [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+import java.util.function.Supplier;
+
+public class SpawnerManager extends ServerFeature {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SpawnerManager.class);
+
+    private static SpawnerManager INSTANCE = new SpawnerManager();
+    private DataStore dataStore;
+
+    // Defer loading so the config and registry are initialized
+    private final LazyValue<List<EntityType<?>>> randomEntities = new LazyValue<>(() -> {
```
Comment (desht) [inline common/src/main/java/dev/ftb/packcompanion/features/spawners/SpawnerManager.java]: Would `Suppliers.memoize()` work here?

**Review by desht — APPROVED** [review]
Looks good!

**Pull request merged**